prow/sidecar/run_test.go

```diff
@@ -148,6 +148,152 @@ func TestWait(t *testing.T) {
 	}
 }
 
+func TestWaitParallelContainers(t *testing.T) {
+	aborted := strconv.Itoa(entrypoint.AbortedErrorCode)
+	const (
+		pass                 = "0"
+		fail                 = "1"
+		missingMarkerTimeout = time.Second
+	)
+	cases := []struct {
+		name         string
+		markers      []string
+		abort        bool
+		pass         bool
+		accessDenied bool
+		missing      bool
+		failures     int
+	}{
+		{
+			name:    "pass, not abort when 1 item passes",
+			markers: []string{pass},
+			pass:    true,
+		},
+		{
+			name:    "pass when all items pass",
+			markers: []string{pass, pass, pass},
+			pass:    true,
+		},
+		{
+			name:     "fail, not abort when 1 item fails",
+			markers:  []string{fail},
+			failures: 1,
+		},
+		{
+			name:     "fail when any item fails",
+			markers:  []string{pass, fail, pass},
+			failures: 1,
+		},
+		{
+			name:     "abort and fail when 1 item aborts",
+			markers:  []string{aborted},
+			abort:    true,
+			failures: 1,
+		},
+		{
+			name:     "abort when any item aborts",
+			markers:  []string{pass, aborted, fail},
+			abort:    true,
+			failures: 2,
+		},
+		{
+			name:     "fail when marker does not exist",
+			markers:  []string{pass},
+			missing:  true,
+			failures: 1,
+		},
```

Comment:
Please also add a test case to ensure that the exit code we use to indicate a previous step failed is not counted as an additional failure.

Reply:
Would it make sense to test this for containers in parallel?  There isn't really any previous step since there isn't a deterministic ordering of containers in this case right?